src/compiler/diagnosticMessages.json

```diff
@@ -1304,6 +1304,34 @@
         "category": "Error",
         "code": 2484
     },
+    "The left-hand side of a 'for...of' statement cannot be a previously defined constant.": {
+        "category": "Error",
+        "code": 2485
+    },
+    "The left-hand side of a 'for...in' statement cannot be a previously defined constant.": {
+        "category": "Error",
+        "code": 2486
+    },
+    "Invalid left-hand side in 'for...of' statement.": {
+        "category": "Error",
+        "code": 2487
+    },
+    "The right-hand side of a 'for...of' statement must have a '[Symbol.iterator]()' method that returns an iterator.": {
+        "category": "Error",
+        "code": 2488
+    },
+    "The iterator returned by the right-hand side of a 'for...of' statement must have a 'next()' method.": {
+        "category": "Error",
+        "code": 2489
+    },
+    "The object returned by the next method of the iterator must have a 'value' property.": {

```

Comment:
You're totally right about `next`. I will fix that.

I see your point. I guess unqualified noun phrases should be indefinite. So in a possessive chain of noun phrases, the tail should be indefinite, but all the preceding ones can be definite.